assembly/src/parsers/ast/context.rs

```diff
@@ -320,6 +369,42 @@ impl ParserContext {
     }
 }
 
+#[derive(PartialEq)]
+enum AsmSourceState {
+    Empty,
+    Comment,
+    Func,
+}
+
+fn parse_new_state(line: &str) -> AsmSourceState {
+    if line.starts_with(DOC_COMMENT_PREFIX) {
+        AsmSourceState::Comment
+    } else if line.starts_with(FUNC_PREFIX) {
+        AsmSourceState::Func
+    } else {
+        AsmSourceState::Empty
+    }
+}
+
+fn remove_prefix(prefix: &str, line: &str) -> String {
+    let mut stripped_line = line.strip_prefix(prefix).unwrap();
+
+    if stripped_line.starts_with(' ') {
+        stripped_line = stripped_line.strip_prefix(' ').unwrap();
+    }
+    if stripped_line.ends_with(' ') {
+        stripped_line = stripped_line.strip_suffix(' ').unwrap();
+    }
+
+    String::from(stripped_line)
+}
```

Comment:
Let's add comments to this function because it wasn't immediately clear to me what it is trying to accomplish. Also, I think there is probably a simpler way to write this (assuming I understood it correctly). Maybe something like this:
```Rust
fn remove_prefix(prefix: &str, line: &str) -> String {
    let trimmed_line = line.trim();
    debug_assert!(trimmed_line.len() > 2);
    String::from(trimmed_line[2..].trim())
}
```